loader/src/main/kotlin/net/weavemc/loader/util/Utility.kt

```diff
@@ -138,7 +140,61 @@ internal fun fatalError(message: String): Nothing {
         /* messageType = */ JOptionPane.ERROR_MESSAGE
     )
 
-    exitProcess(-1)
+    exit(-1)
+    throw IllegalStateException("Exiting.")
+}
+
+/**
+ * Exits the JVM with the given error code, escaping any SecurityManager
+ * in place (looking at you Froge).
+ *
+ * @param errorCode the error code to exit with
+ */
+fun exit(errorCode: Int) {
+	runCatching {
+        val clazz = Class.forName("java.lang.Shutdown")
+        clazz.getDeclaredMethod("exit", Int::class.javaPrimitiveType).apply {
+            isAccessible = true
+        }.invoke(null, errorCode)
+    }.onFailure { e0 ->
+		runCatching {
+            exitRuntime(errorCode)
+		}.onFailure { e1 ->
+            if (getJavaVersion() <= 19) { // beware of class removal
+                AccessController.doPrivileged(PrivilegedAction<Void> {
+                    exitRuntime(errorCode)
+                    null
+                })
+            } else {
+                // this'll exit alright, but it's not pretty
+				e1.addSuppressed(e0)
+                throw RuntimeException("Exitting the JVM, no errors to report here.", e1)
+            }
+        }
+    }
+}
```

Comment:
format please :)

Reply:
whoops, that was done in vim really quickly lmao